Sources/SwiftLexicalScopes/Scope.swift

```diff
@@ -0,0 +1,185 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import SwiftSyntax
+
+extension SyntaxProtocol {
+  /// Scope at the syntax node. Could be inherited from parent or introduced at the node.
+  var scope: Scope? {
+    switch self.syntaxNodeType {
+    case is SourceFileSyntax.Type:
+      FileScope(syntax: self)
+    default:
+      parent?.scope
+    }
+  }
+}
+
+/// Provide common functionality for specialized scope implementatations.
+protocol Scope {
+  /// The parent of this scope.
+  var parent: Scope? { get }
+
+  /// Syntax node that introduces this protocol.
+  var sourceSyntax: SyntaxProtocol { get }
+}
```

Comment:
As far as I can tell, `Scope` isn't doing anything any more. Can we remove it, and put the various operations on `SyntaxProtocol` directly?

Reply:
It was a leftover from the original API structure I wanted to implement. I moved now all the query methods to `SyntaxProtocol` and removed parts related to the `Scope` protocol.